modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/ScheduleModal.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayAlert from '@clayui/alert';
+import ClayButton from '@clayui/button';
+import ClayDatePicker from '@clayui/date-picker';
+import {isAfter} from 'date-fns';
+import {getOpener} from 'frontend-js-web';
+import PropTypes from 'prop-types';
+import React, {useEffect, useState} from 'react';
+
+const SCHEDULE_EVENT_NAME = 'scheduleKBArticle';
+export default function ScheduleModal({
+	namespace,
+	scheduledDate: initialScheduleDate,
+}) {
+	const [scheduledDate, setScheduledDate] = useState(initialScheduleDate);
+	const [invalidDate, setInvalidDate] = useState(false);
+
+	const closeModal = () => {
+		getOpener().Liferay.fire('closeModal', {
+			id: 'scheduleKBArticleDialog',
+		});
+	};
+
+	const handleScheduleButtonClick = () => {
+		const openerWindow = getOpener();
+
+		const scheduleDateInput = openerWindow.document.getElementById(
+			`${namespace}scheduleDate`
+		);
+		scheduleDateInput.value = scheduledDate;
+
+		getOpener().Liferay.fire(SCHEDULE_EVENT_NAME);
+		closeModal();
+	};
+
+	useEffect(() => {
+		if (
+			isAfter(Date.parse(scheduledDate), Date.now()) ||
+			(Number.isNaN(Date.parse(scheduledDate)) && !scheduledDate)
+		) {
+			setInvalidDate(false);
+		}
+		else {
+			setInvalidDate(true);
+		}
+	}, [scheduledDate]);
+
+	return (
+		<div className="schedule-modal">
+			<div className="container-fluid p-4">
+				<p className="text-secondary">
+					{Liferay.Language.get('set-date-and-time-for-publication')}
+				</p>
+
+				{invalidDate === true ? (
```

Comment:
All this section can be simplified so there is no need to have two different ClayDatePickers to maintain: 
```
<div className={classnames({'has-error': invalidDate})}>  --> use the className utility
<ClayDatePicker .../>
{invalidDate && <ClayAlert className="mt-1" .../>}
```